testing/test.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import json
+import shutil
+import subprocess
+import shlex
+import argparse
+
+# Set up argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument(dest="path_to_exercise", metavar="PATH_TO_EXERCISE", type=str, help="the path to the exercise")
+args = parser.parse_args()
+
+path_to_exercise = args.path_to_exercise
+assert path_to_exercise 

```

Comment:
Since `path_to_exercise` is a required argument, it cannot be `None` and an empty string would be caught by the following assertion, so this line can be removed.